x-pack/plugins/infra/common/log_sources/resolved_log_source_configuration.ts

```diff
@@ -117,7 +118,7 @@ const resolveRuntimeMappings = (indexPattern: IndexPattern): estypes.MappingRunt
           ? {
               script: {
                 lang: 'painless', // required in the es types
-                source: runtimeFieldSpec.script.source,
+                source: (runtimeFieldSpec.script as estypes.InlineScript).source,
```

Comment:
Now that the type of `runtimeFieldSpec` supports more than just inline scripts, don't we need to support those here as well? This cast looks unsafe to me, because it might be an indexed script according to the type.

Reply:
I had to cast it to avoid a TS error

<img width="967" alt="Screenshot 2021-08-31 at 17 18 17" src="https://user-images.githubusercontent.com/2854616/131539334-4bc111cc-5012-4d23-957e-c103eb760752.png">

Reply:
Right, I meant that we need to support `IndexedScript` in addition to `InlineScript`, which we previously didn't because the runtime field type wouldn't allow for it. I can contribute a small commit tomorrow if you like.

Reply:
I am not sure that it's the issue. Typescript does not know if `runtimeFieldSpec.script` is a `string`, an `IndexedScript` or an `InlineScript` until we cast it. But if you think it something else let me know 👍